samples/nrf_desktop/src/hw_interface/motion_simulated.c

```diff
@@ -0,0 +1,218 @@
+	u32_t cur_time = k_uptime_get_32();
+	int x_new, y_new;
+
+	int start = (cur_time - start_time) / move_time;
+	int end;
+
+	if (start >= ARRAY_SIZE(coords)) {
+		start_time += move_time * (ARRAY_SIZE(coords));
+		start = (cur_time - start_time) / move_time;
+	}
+
+	if (start == ARRAY_SIZE(coords) - 1) {
+		end = 0;
+	} else {
+		end = start + 1;
+	}
+
+	x_new = coords[start][COORD_X]
+		+ (coords[end][COORD_X] - coords[start][COORD_X])
+		* ((int)cur_time - (int)start_time - start * move_time)
+		/ move_time;
+	y_new = coords[start][COORD_Y]
+		+ ((coords[end][COORD_Y] - coords[start][COORD_Y])
+		* ((int)cur_time - (int)start_time - start * move_time))
+		/ move_time;
+
+	motion_event_send(x_new - x_cur, y_new - y_cur);
+
+	x_cur = x_new;
+	y_cur = y_new;
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_hid_report_subscription_event(eh)) {
+		const struct hid_report_subscription_event *event =
+			cast_hid_report_subscription_event(eh);
+
+		if (event->report_type == TARGET_REPORT_MOUSE) {
+			static u8_t peer_count;
+
+			if (event->enabled) {
+				__ASSERT_NO_MSG(peer_count < UCHAR_MAX);
+				peer_count++;
+			} else {
+				__ASSERT_NO_MSG(peer_count > 0);
+				peer_count--;
+			}
+
+			bool new_state = (peer_count != 0);
+			bool old_state = atomic_set(&connected, new_state);
+
+			if (old_state != new_state && new_state) {
+				if (atomic_get(&state) == STATE_FETCHING) {
+					start_time = k_uptime_get_32();
+					x_cur = coords[0][COORD_X];
+					y_cur = coords[0][COORD_Y];
+					generate_motion_event();
+				}
+			}
+		}
+		return false;
+	}
+
+	if (is_hid_report_sent_event(eh)) {
+		const struct hid_report_sent_event *event =
+			cast_hid_report_sent_event(eh);
+
+		if ((event->report_type == TARGET_REPORT_MOUSE) &&
+				(atomic_get(&state) == STATE_FETCHING)) {
+			generate_motion_event();
+		}
+
+		return false;
+	}
+
+	if (is_module_state_event(eh)) {
+		struct module_state_event *event = cast_module_state_event(eh);
+
+		if (check_state(event, MODULE_ID(board), MODULE_STATE_READY)) {
+			atomic_set(&state, STATE_IDLE);
+			LOG_INF("Simulated motion: ready");
+		}
+
+		return false;
+	}
+
+	if (is_power_down_event(eh)) {
+		atomic_set(&state, STATE_SUSPENDED);
+
+		return false;
+	}
+
+	if (is_wake_up_event(eh)) {
+		atomic_set(&state, STATE_IDLE);
+
+		return false;
+	}
+
+	/* If event is unhandled, unsubscribe. */
+	__ASSERT_NO_MSG(false);
+
+	return false;
+}
+
+EVENT_LISTENER(MODULE, event_handler);
+EVENT_SUBSCRIBE(MODULE, power_down_event);
+EVENT_SUBSCRIBE(MODULE, module_state_event);
+EVENT_SUBSCRIBE(MODULE, wake_up_event);
+EVENT_SUBSCRIBE(MODULE, hid_report_sent_event);
+EVENT_SUBSCRIBE(MODULE, hid_report_subscription_event);
+
+
+static int start_motion(const struct shell *shell, size_t argc, char **argv)
+{
+	if (!atomic_cas(&state, STATE_IDLE, STATE_FETCHING)) {
+		shell_print(shell, "Simulated motion inactive"
+				    " or already fetching");
+		return 0;
+	}
+
+	if (atomic_get(&connected)) {
```

Comment:
this won't be needed if you use what I proposed

Reply:
done